docs/examples/shading/plot_martinez_shade_loss.py

```diff
@@ -0,0 +1,266 @@
+    solar_pos["apparent_zenith"],
+    solar_pos["azimuth"],
+)  # unpack for better readability
+
+tracking_result = pvlib.tracking.singleaxis(
+    apparent_zenith=solar_apparent_zenith,
+    apparent_azimuth=solar_azimuth,
+    axis_tilt=axis_tilt,
+    axis_azimuth=axis_azimuth,
+    max_angle=(-90 + cross_axis_tilt, 90 + cross_axis_tilt),  # (min, max)
+    backtrack=False,
+    gcr=gcr,
+    cross_axis_tilt=cross_axis_tilt,
+)
+
+tracker_theta, aoi, surface_tilt, surface_azimuth = (
+    tracking_result["tracker_theta"],
+    tracking_result["aoi"],
+    tracking_result["surface_tilt"],
+    tracking_result["surface_azimuth"],
+)  # unpack for better readability
+
+# Calculate the tracking angles
+rotation_angle = pvlib.tracking.singleaxis(
+    solar_apparent_zenith,
+    solar_azimuth,
+    axis_tilt,
+    axis_azimuth,
+    max_angle=(-45, 45),  # (min, max) degrees
+    backtrack=False,
+    gcr=gcr,
+    cross_axis_tilt=cross_axis_tilt,
+)["tracker_theta"]
+
+# Calculate the shade fraction
+shaded_fraction = pvlib.shading.shaded_fraction1d(
+    solar_apparent_zenith,
+    solar_azimuth,
+    axis_azimuth,
+    axis_tilt=axis_tilt,
+    shaded_row_rotation=rotation_angle,
+    shading_row_rotation=rotation_angle,
+    collector_width=width,
+    pitch=pitch,
+    cross_axis_slope=cross_axis_tilt,
+)
+
+# %%
+# Number of shaded blocks
+# -----------------------
+# The number of shaded blocks depends on the module configuration and number
+# of bypass diodes. For example,
+# modules with one bypass diode will behave like one block.
+# On the other hand, modules with three bypass diodes will have three blocks,
+# except for the half-cut cell modules, which will have six blocks; 2x3 blocks
+# where the two rows are along the longest side of the module.
+# We can argue that the dimensions of the system changes when you switch from
+# portrait to landscape, but for this example, we will consider it the same.
+#
+# The number of shaded blocks is calculated by rounding up the shaded fraction
+# by the number of blocks along the shaded length. So let's define the number
+# of blocks for each module configuration:
+#
+# - 1 bypass diode: 1 block
+# - 3 bypass diodes: 3 blocks (in portrait; in landscape, it would be 1)
+# - 3 bypass diodes half-cut cells:
+#   - 2 blocks in portrait
+#   - 3 blocks in landscape
+#
+# .. figure:: ../../_images/PV_module_layout_cesardd.jpg
+#    :align: center
+#    :width: 75%
+#    :alt: Normal and half-cut cells module layouts
+#
+#    Left: common module layout. Right: half-cut cells module layout.
+#    Each module has three bypass diodes. On the left, they connect cell
+#    columns 1-2, 2-3 & 3-4. On the right, they connect cell rows 1-2, 3-4 &
+#    5-6.
+#    *Source: César Domínguez. CC BY-SA 4.0, Wikimedia Commons*
+#
+# .. figure:: ../../_images/Centralized_and_split_PV_junction_boxes_cesardd.jpg
+#    :align: center
+#    :width: 75%
+#    :alt: Centralized and split PV junction boxes
```

Comment:
I'm not sure this diagram is helpful.  I suggest either removing it, or adding some explanation of how junction boxes relate to diodes and blocks.